mantidproject/mantid
# Inelastic Settings option to turn off Load Workspace History

### Description of work
This PR adds an option to the Indirect Settings dialog to load the history of a workspace, for details see #37691 . This will allow for faster workflows as sometimes loading the history can slog the loading algorithm #36769. 
This setting is now available in most indirect/inelastic interfaces that preload data using 'LoadNexusProcessed' algorithm to the ADS (as the ones having data selectors or workspace dialogs for selecting input data). It won't be available for loaders that load raw files (like in diffraction or data reduction interfaces). 



Fixes #37691 . 




#### Further detail of work


### To test:
As there are many affected interfaces, the instructions are slightly tedious, I'm sorry for that. 

- First, make sure to delete the `[Indirect Settings]` and any setting in that group from the `mantidworkbench.ini` file that stores the settings in your machine. (You can find 'mantidworkbench.ini' either in `%APPDATA/../mantidproject` on Windows or in `.config/mantidproject` on MacOS  and Linux.) . Then save the file. 
- Open Mantid Workbench, and then any Indirect/Inelastic interface. Once on the interface, click on the cog icon for the settings, which should be located at the bottom left corner of the interface. The settings dialog will open. 
- On the settings dialog, the options: `Restrict allowed input files by name` and `Load Workspace History` should be ticked by default. Close the dialog. 

Now, for every Inelastic Interface, do the following for each tab (except  on `Calculate Paalman Pings` which is deprecated): 

- Open the settings dialog and tick on the `Load Workspace History` setting. 
- Load sample data for the respective interface (either with a dialog, or from the data selector)
- Once the sample data is loaded onto the ADS, right-click on the workspace and select `Show History`. You should see the history log for that workspace. 
- Clear the ADS. 
- Open the Settings dialog again and untick the `Load Workspace History` setting.
- Load the same sample data as before. 
- Once the sample data is loaded onto the ADS, right-click on the workspace and select `Show History`. You should see only 1 entry on the history log (the load algorithm). 
- Move onto the next interface. 

For the Indirect Interfaces, it is only necessary to do the loading history check with the Calibration Data Selector of the `ISIS Energy Transfer` and `ISIS Diagnostics` tabs (the one allowing to load calibration files) and with the Transmission tab. 

Finally, build the docs and check that the `Interfaces/Indirect Settings` section contains and entry for the load history settings. Check that is well formatted and understandable. 


*A release note has been added and the documentation entry for settings dialog updated*



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
